src/lib.rs

```diff
@@ -19,10 +20,57 @@
 #[cfg(feature = "alloc")]
 extern crate alloc;
 
-#[doc(hidden)]
-pub mod _export {
-    /// A re-export of core::*
-    pub mod _core {
-        pub use core::*;
+mod error;
+mod iter;
+mod parse;
+
+#[cfg(all(feature = "alloc", not(feature = "std")))]
+use alloc::vec::Vec;
+
+use crate::parse::FromHex as _;
+
+#[rustfmt::skip]                // Keep public re-exports separate.
+#[doc(inline)]
+pub use self::{
+    error::{
+        DecodeToArrayError, DecodeToBytesError, InvalidCharError,
+        InvalidLengthError, OddLengthStringError, 
+    },
```

Comment:
FTR I think these re-exports are alright. I expect `hex::WhateverError` to be common and I like it.